### PR TITLE
Load entities and policies from json file

### PR DESCRIPTION
**Pull request by cleissonbarbosa**
**Reason**: Unable to load a large amount of entities (30,000 - 4.5MB) from the `/data` endpoint

**What PR does**: Adds the possibility to load entities and policies from a json file at rocket startup

Use the `-d` or `--data` option and inform the path of the json file you want to load in cedar's memory, it is also possible to use a `DATA` environment variable

examples:

```bash
cargo run -- -d "./examples/data.json"
```

```bash
cargo run -- --data="./examples/data.json"
```

```bash
export DATA="./examples/data.json"
cargo run
```
---

Use the  `--policies` option and inform the path of the json file you want to load in cedar's memory, it is also possible to use a `POLICIES` environment variable

examples:

```bash
cargo run -- --policies="./examples/policies.json"
```

```bash
export POLICIES="./examples/policies.json"
cargo run
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -95,6 +95,9 @@ Cedar Agent configuration is available using environment variables and command l
 - The log level to filter logs. Defaults to `info`.  
   `LOG_LEVEL` environment variable.  
   `--log-level`, `-l` command line argument.
+- Load data from json file. Defaults to `None`.  
+  `DATA` environment variable.
+  `--data`, `-d` command line argument.
```
Comment (omer9564) [inline README.md:100]: missing policy file configuration

**Review by omer9564 — CHANGES_REQUESTED** [review]
Looks good in general, the only major changes needed here is missing tests

**Comment by cleissonbarbosa** [comment]
@omer9564 Thank you very much for the review, in addition to the changes you requested, I changed the word policy to policies in the config.

File: tests/services/data_tests.rs [hunk 2]
```diff
@@ -18,3 +24,28 @@ async fn memory_tests() {
     let entities = store.get_entities().await;
     assert_eq!(entities.len(), 0);
 }
+
+#[tokio::test]
+async fn test_load_entities_from_file() {
+    let temp_file_path = Arc::new(PathBuf::from("test_entities.json"));
```
Comment (omer9564) [inline tests/services/data_tests.rs:30]: you could also use the data.json from the examples directory

File: tests/services/data_tests.rs [hunk 3]
```diff
@@ -18,3 +24,28 @@ async fn memory_tests() {
...
+        }
+    }]"#;
+    let mut temp_file = File::create(&*temp_file_path.clone()).unwrap();
+    temp_file.write_all(test_data.as_bytes()).unwrap();
+
+    let entities = load_entities_from_file(temp_file_path.clone().to_path_buf()).await.unwrap();
+    assert_eq!(entities.len(), 1);
+
+    std::fs::remove_file(temp_file_path.clone().to_path_buf()).unwrap();
+}
```
Comment (omer9564) [inline tests/services/data_tests.rs:51]: missing new line at end of file

File: tests/services/policies_tests.rs [hunk 4]
```diff
@@ -82,3 +88,20 @@ async fn memory_tests() {
         .policy(&PolicyId::from_str("test").unwrap())
         .is_none());
 }
+
+#[tokio::test]
+async fn test_load_policies_from_file() {
+    let temp_file_path = Arc::new(PathBuf::from("test_policies.json"));
```
Comment (omer9564) [inline tests/services/policies_tests.rs:94]: you could also use the policies.json from the examples directory

**Review by omer9564 — CHANGES_REQUESTED** [review]
two suggestion comments ( might make the code simpler )
one missing new line again 😅 , Besides that it looks great 💪

**Comment by cleissonbarbosa** [comment]
> two suggestion comments ( might make the code simpler ) one missing new line again 😅 , Besides that it looks great 💪

😅 thank you very much for the suggestion.

**Comment by danielbass37** [comment]
@cleissonbarbosa PMed you in the community <3

**Comment by cleissonbarbosa** [comment]
@orweis @danielbass37 Thank you guys 💜

**Review by omer9564 — APPROVED** [review]
LGTM

**Pull request merged**